tests/test_context_managers.py

```diff
@@ -0,0 +1,91 @@
+import asyncio
+import contextvars
+import uuid
+
+
+session_id_var = contextvars.ContextVar("session_id", default=str(uuid.uuid4()))
+
+
+class ExampleSession:
+    def __init__(self):
+        # print("Initializing the context...")
+        pass
+
+    def __enter__(self):
+        self.token = session_id_var.set(str(uuid.uuid4()))
+
+    def __exit__(self, exc_type, exc_value, exc_tb):
+        session_id_var.reset(self.token)
+
+    async def __aenter__(self):
+        self.token = session_id_var.set(str(uuid.uuid4()))
+
+    async def __aexit__(self, exc_type, exc_value, exc_tb):
+        session_id_var.reset(self.token)
+
+
+def simulated_llm_call():
```

Comment:
If you removed the default value - I'd expect to see an explicit set here (or in an outer scope) before get is possible.

Reply:
Note sure I follow. Is it to set uuids explicitly by having it as an input to the session object?